dds/DCPS/Observer.h

```diff
@@ -84,18 +102,40 @@ class OpenDDS_Dcps_Export Observer
   virtual void on_disassociated(DDS::DataReader_ptr, const GUID_t& /* writerId */) {}
 
   // Group 3: Sample sent, received, read, taken
-  virtual void on_sample_sent(DDS::DataWriter_ptr, const Sample&, const ValueWriterDispatcher&) {}
-  virtual void on_sample_received(DDS::DataReader_ptr, const Sample&, const ValueWriterDispatcher&) {}
-  virtual void on_sample_read(DDS::DataReader_ptr, const Sample&, const ValueWriterDispatcher&) {}
-  virtual void on_sample_taken(DDS::DataReader_ptr, const Sample&, const ValueWriterDispatcher&) {}
+  virtual void on_sample_sent(DDS::DataWriter_ptr, const Sample&) {}
+  virtual void on_sample_received(DDS::DataReader_ptr, const Sample&) {}
+  virtual void on_sample_read(DDS::DataReader_ptr, const Sample&) {}
+  virtual void on_sample_taken(DDS::DataReader_ptr, const Sample&) {}
 
-  virtual ~Observer();
+  virtual ~Observer() {}
```

Comment:
These are typically in the .cpp so that the vtable (and related stuff) get generated in this object file.  Not sure that's so important these days.